### ISSUE TITLE
[Bug] Incomptability with TypeScript 5.4

### ISSUE THREAD

**Issue by benasher44** [motiondivision/motionone#252]
**1. Describe the bug**

```
node_modules/@motionone/dom/types/animate/types.d.ts:5:18 - error TS2430: Interface 'CSSStyleDeclarationWithTransform' incorrectly extends interface 'Omit<CSSStyleDeclaration, "direction" | "transition">'.
  Types of property 'x' are incompatible.
    Type 'number' is not assignable to type 'string'.

5 export interface CSSStyleDeclarationWithTransform extends Omit<CSSStyleDeclaration, "direction" | "transition"> {
```

**2. IMPORTANT: Provide a CodeSandbox reproduction of the bug**

If it's really necessary, I can make one. but the issue is that CSSStyleDeclaration was updated to have x, y, z b string instead of number.

**3. Steps to reproduce**

1. Create a TS project with TS5.4
2. Add @motionone/dom
3. Run tsc

**4. Expected behavior**

It compiles without issue.

**5. Video or screenshots**

See above error snippet

**6. Browser details**

N/A

**Comment by marcolmosdev** [comment]
Any solution? Thanks :)

**Comment by mattgperry** [comment]
This should be fixed in `10.18`!
